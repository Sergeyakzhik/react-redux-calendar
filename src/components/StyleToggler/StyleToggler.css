.btn-group label {
  color: black;
  background-color: white;
  width: 3rem;
  height: 2.7rem;
  margin-bottom: 0;
  margin-top: 0.2rem;
}

.style-toggler {
  margin-left: 4rem;
  padding: 0;
}

.style-toggler label {
  cursor: pointer;
}

.style-option {
  width: 100%;
  height: 100%;
  margin: 0;
}

.style-option:first-of-type {
  border-top-left-radius: 5px!important;
  border-bottom-left-radius: 5px!important;
}

.style-option:last-of-type {
  border-top-right-radius: 5px;
  border-bottom-right-radius: 5px;
}

.btn-group label.style1 {
  background-image: linear-gradient(to bottom right, black, rgb(82, 82, 82));
}

.btn-group label.style2 {
  background-image: linear-gradient(to bottom right, rgb(239, 253, 192), rgb(200, 233, 252));
}

input[type="radio"] {
  display: none;
}
