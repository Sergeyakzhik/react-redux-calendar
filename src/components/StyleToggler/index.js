import React from 'react';
import './style.css';

let StyleToggler = props => (
  <div className="btn-group btn-group-toggle style-toggler radio">
    <input type="radio" name="style-toggler" autoComplete="off" id="style1" value="style1" onChange={props.onStyleTogglerClick} checked={props.style === "style1"}/>
    <label htmlFor="style1"  className="btn btn-secondary style-option style1"></label>
    <input type="radio" name="style-toggler" autoComplete="off" id="style2" value="style2" onChange={props.onStyleTogglerClick} checked={props.style === "style2"}/>
    <label htmlFor="style2" className="btn btn-secondary style-option style2"></label>
  </div>
)

export default StyleToggler;
