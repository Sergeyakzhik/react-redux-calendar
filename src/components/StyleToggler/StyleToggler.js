import React from 'react';
import './StyleToggler.css';

let StyleToggler = props => (
  <div className="btn-group btn-group-toggle style-toggler radio">
    <input type="radio" name="style-toggler" autoComplete="off" id="style1" value="style1" onChange={props.onStyleTogglerClick} checked={props.style === "style1"}/>
    <label for="style1"  className="btn btn-secondary style-option style1"></label>
    <input type="radio" name="style-toggler" autoComplete="off" id="style2" value="style2" onChange={props.onStyleTogglerClick} checked={props.style === "style2"}/>
    <label for="style2" className="btn btn-secondary style-option style2"></label>
    <input type="radio" name="style-toggler" autoComplete="off" id="style3" value="style3" onChange={props.onStyleTogglerClick} checked={props.style === "style3"}/>
    <label for="style3" className="btn btn-secondary style-option style3"></label>
  </div>
)

export default StyleToggler;
