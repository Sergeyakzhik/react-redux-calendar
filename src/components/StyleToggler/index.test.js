import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

import StyleToggler from './index';

Enzyme.configure({ adapter: new Adapter() })

describe('StyleToggler', () => {
  let props = {
    style: {},
    onStyleTogglerClick: () => {}
  }
  let component = shallow(<StyleToggler {...props} />);

  it('renders StyleToggler component', () => {
    expect(component).toMatchSnapshot();
  })
})
