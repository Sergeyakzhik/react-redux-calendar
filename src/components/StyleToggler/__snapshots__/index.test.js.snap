// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyleToggler render StyleToggler component 1`] = `
<div
  className="btn-group btn-group-toggle style-toggler radio"
>
  <input
    autoComplete="off"
    checked={false}
    id="style1"
    name="style-toggler"
    onChange={[Function]}
    type="radio"
    value="style1"
  />
  <label
    className="btn btn-secondary style-option style1"
    htmlFor="style1"
  />
  <input
    autoComplete="off"
    checked={false}
    id="style2"
    name="style-toggler"
    onChange={[Function]}
    type="radio"
    value="style2"
  />
  <label
    className="btn btn-secondary style-option style2"
    htmlFor="style2"
  />
</div>
`;
